File: src/leetcode/2025/12/Solution326.cpp
#include <iostream>

class Solution326
{
public:
    bool isPowerOfThree(int n)
    {
        if (n < 1)
        {
            return false;
        }
        while (n % 3 == 0)
        {
            n /= 3;
        }
        return n == 1;
    }
};

int main()
{
    Solution326 solution;

    std::cout << solution.isPowerOfThree(27) << std::endl; // true
    std::cout << solution.isPowerOfThree(0) << std::endl;  // false
    std::cout << solution.isPowerOfThree(9) << std::endl;  // true
    std::cout << solution.isPowerOfThree(45) << std::endl; // false
    return 0;
}

File: src/leetcode/2025/12/Solution326.java
class Solution326 {
    public boolean isPowerOfThree(int n) {
        if (n < 1) {
            return false;
        }
        while (n % 3 == 0) {
            n /= 3;
        }
        return n == 1;
    }

    public static void main(String[] args) {
        Solution326 solution = new Solution326();
        System.out.println(solution.isPowerOfThree(27)); // true
        System.out.println(solution.isPowerOfThree(0)); // false
        System.out.println(solution.isPowerOfThree(1)); // true
        System.out.println(solution.isPowerOfThree(-1)); // false
        System.out.println(solution.isPowerOfThree(9)); // true
        System.out.println(solution.isPowerOfThree(45)); // false
    }
}